src/TypedStateMachines/Actions/EdgesBuilder.php

```diff
@@ -82,21 +103,32 @@ function($mapping) use ($state) {
         return $this;
     }
 
+    /**
+     * Specify the Transition of the Edge.
+     *
+     * @param Transition $transition
+     * @return EdgesBuilder
+     */
     public function transition(Transition $transition): EdgesBuilder
     {
         $this->buildEdges();
         $this->current_transition = $transition;
         return $this;
     }
 
+    /**
+     * Build the current Transition/source/target into the edges array.
+     *
+     * @return void
+     */
     private function buildEdges()
     {
         if ($this->current_transition == null) {
             return;
         }
 
         if (! $this->current_mappings) {
```

Comment:
Current mappings is an array 
so wouldnt it be clearer to use empty() instead of !

Reply:
I don't want to make that change. In some cases, it can be null.

I'm not 100% sure how, but looking at the sources() method, it can be null, otherwise it would be throwing a LogicException

Reply:
Because we first call the transition() method when we start building, it will set the current_mappings to null.